From 9157ace84027ed231067ff6c3e6ab7a6384d1c6f Mon Sep 17 00:00:00 2001
From: Jingkui Wang <jkwang@google.com>
Date: Thu, 14 Jun 2018 15:29:32 -0700
Subject: [PATCH 1/1] Patch lxd to enable tcp KeepAlive

This will stop lxc from block indefinitely when there is network change.
---
 shared/network.go | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/shared/network.go b/shared/network.go
index 2ccb7dc8..f47f75ce 100644
--- a/shared/network.go
+++ b/shared/network.go
@@ -31,6 +31,10 @@ func RFC3493Dialer(network, address string) (net.Conn, error) {
 		if err != nil {
 			continue
 		}
+		if tc, ok := c.(*net.TCPConn); ok {
+			tc.SetKeepAlive(true)
+			tc.SetKeepAlivePeriod(3 * time.Second)
+		}
 		return c, err
 	}
 	return nil, fmt.Errorf("Unable to connect to: " + address)
-- 
2.18.0.rc1.244.gcf134e6275-goog

